packages/jaeger-ui/src/utils/helpers/ReactShallowRenderer.js

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2023 The Jaeger Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
```

Comment:
```suggestion
Adopted from https://thesametech.com/snapshot-testing-in-rtl/
Copyright (c) 2023 Ildar Sharafeev

Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice (including the next paragraph) shall be included in all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
```

I reached out to @sr-shifu for permission to use the code from the blog via MIT license. We cannot merge without that permission.

An alternative is to avoid using snapshots and test user-facing behavior of component instead, along the lines of RTL recommendation.

Reply:
Licensing is important. Let's wait for a few days (2-3) for his response.
In case we are denied or not answered, I would help the project in migration from snapshot testing altogether.

Reply:
Looks good guys! Thank you for taking care of licensing aspects of the software you are using! 

Hopefully, you will find this tool helpful!

Reply:
Thankyou @sr-shifu for allowing us, and for writing that wonderful article 🚀